modules/fundamental/src/advisory/endpoints/mod.rs

```diff
@@ -1,26 +1,34 @@
+#[cfg(test)]
+mod test;
+
 use crate::advisory::service::AdvisoryService;
 use crate::Error;
 use actix_web::{get, post, web, HttpResponse, Responder};
 use futures_util::TryStreamExt;
 use std::str::FromStr;
+use std::sync::Arc;
 use tokio_util::io::ReaderStream;
+use trustify_auth::authenticator::Authenticator;
 use trustify_common::db::query::Query;
 use trustify_common::db::Database;
 use trustify_common::hash::HashKey;
 use trustify_common::model::Paginated;
+use trustify_infrastructure::app::new_auth;
 use trustify_module_ingestor::service::{Format, IngestorService};
 use trustify_module_storage::service::StorageBackend;
 use utoipa::{IntoParams, OpenApi};
 
-pub fn configure(config: &mut web::ServiceConfig, db: Database) {
+pub fn configure(config: &mut web::ServiceConfig, db: Database, auth: Option<Arc<Authenticator>>) {
     let advisory_service = AdvisoryService::new(db);
 
-    config
-        .app_data(web::Data::new(advisory_service))
-        .service(all)
-        .service(get)
-        .service(upload)
-        .service(download);
+    config.app_data(web::Data::new(advisory_service)).service(
+        web::scope("/api/v1/advisory")
+            .wrap(new_auth(auth))
```

Comment:
Why do we have to do this here? Can't we inject the auth in the code that calls `configure`? I'd like to keep the api paths DRY.

Reply:
I am not sure how that should work. The scope needs to be wrapped with the auth-middleware. Nothing else.